recipes/liquid-dsp/all/conanfile.py

```diff
@@ -0,0 +1,178 @@
+    description = "Digital signal processing library for software-defined radios "
+    topics = ("conan", "dsp", "sdr", "liquid-dsp")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jgaeddert/liquid-dsp"
+    license = ("MIT",)
+    settings = "os", "arch", "build_type", "compiler"
+    exports_sources = ["generate_link_library.bat", "patches/**"]
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "simdoverride": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "simdoverride": False,
+    }
+
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _libname(self):
+        if self.settings.os == "Windows":
+            return "libliquid"
+        return "liquid"
+
+    @property
+    def _target_name(self):
+        if self.settings.os == "Macos":
+            if not self.options.shared:
+                return "libliquid.ar"
+            return "libliquid.dylib"
+        if not self.options.shared:
+            return "libliquid.a"
+        return "libliquid.so"
+
+    @property
+    def _lib_pattern(self):
+        if self.settings.os == "Macos" and not self.options.shared:
+            return "libliquid.a"
+        if self.settings.os != "Windows":
+            return self._target_name
+        return "libliquid.lib"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os != "Windows":
+            if not self.options.fPIC:
+                raise ConanInvalidConfiguration("This library hardcodes fPIC")
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+        if self.settings.compiler == "Visual Studio":
+            self.build_requires("mingw-w64/8.1")
+            self.build_requires("automake/1.16.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob("liquid-dsp-*")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
+        if self.settings.os == "Windows":
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+
+    def _gen_link_library(self):
+        if self.settings.compiler != "Visual Studio" or (not self.options.shared):
+            return
+        self.run("cmd /c generate_link_library.bat")
+        with tools.chdir(self._source_subfolder):
+            self.run(
+                f"""{os.getenv("AR")} /def:libliquid.def /out:libliquid.lib """
```

Comment:
```suggestion
                f"""{os.getenv("AR")} /def:libliquid.def /out:libliquid.lib """
```
Python string-format is not allowed, due Conan compatibility with 2.7. Remove all occurrences.